### PR TITLE
Refactor build to use YAML templates

### PR DESCRIPTION
**Pull request by tomkerkhove**
Refactor build to use YAML templates

### CODE REVIEW

File: build/myget-release.yml [hunk 1]
```diff
@@ -160,4 +102,4 @@ stages:
             inputs:
               artifact: 'Build'
               path: '$(Build.SourcesDirectory)'
-          - template: build/nuget/publish-preview-package.yml@templates
```
Comment (tomkerkhove) [inline build/myget-release.yml]: I'll delete the old file once merged.

File: build/ci-build.yml [hunk 2]
```diff
@@ -67,18 +75,10 @@ stages:
...
-          - task: DotNetCoreInstaller@0
-            displayName: 'Import .NET Core SDK ($(DotNet.Sdk.Version))'
-            inputs:
-              version: '$(DotNet.Sdk.Version)'
-          - task: DotNetCoreCLI@2
-            displayName: 'Run unit tests'
-            inputs:
-              command: test
-              projects: 'src/**/$(Project).Tests.Unit.csproj'
-              arguments: '--configuration $(Build.Configuration)'
```
Comment (tomkerkhove) [inline build/ci-build.yml]: Ha no version prefix.

What would you do @stijnmoreels : Do we add something fixed and don't care or see if we can add ifs?

Comment (stijnmoreels) [reply build/ci-build.yml]: Version of the build nuget package? Or from the .net SDK? I think you talk about the variable that has 'preview' as value?

Comment (tomkerkhove) [reply build/ci-build.yml]: Other builds use --version-prefix, which is now part of the task but not the CI build.

Comment (stijnmoreels) [reply build/ci-build.yml]: Yeah, could we have optional parameters in the template? So we can ignore the version prefix in the CI build?

Comment (tomkerkhove) [reply build/ci-build.yml]: We could, we'd just need to figure out how it works in terms of passing the parameter or not.

Should have some time later on to figure that out unless you can already do some research on this ofcourse.

Comment (tomkerkhove) [reply build/ci-build.yml]: Resolving as we're tracking this in https://github.com/arcus-azure/azure-devops-templates/issues/7

**Pull request merged**